### PR TITLE
Android platform directory

### PR DESCRIPTION
**Pull request by eshza**

### CODE REVIEW

File: src/averroes/Main.java [hunk 1]
```diff
@@ -63,29 +68,60 @@ public static void main(String[] args) {
...
 			// Add the organized archives for the application and its
 			// dependencies.
 			TimeUtils.reset();
 			JarFactoryClassProvider provider = new JarFactoryClassProvider();
 			provider.prepareJarFactoryClasspath();
-
-			// Set some soot parameters
+			
+			//Set some soot parameters for android
+            if(AverroesOptions.isAndroid())
```
Comment (karimhamdanali) [inline src/averroes/Main.java]: Please cleanup the indentation for this bock.

File: src/averroes/Main.java [hunk 2]
```diff
@@ -63,29 +68,60 @@ public static void main(String[] args) {
...
+    			if (AverroesOptions.isAndroid()) {	
+    				setupAndroid = SetupAndroid.v();
+    				dummyMain = setupAndroid.getDummyMainMethod(); 
+     			}
+    			else {
+    				Options.v().classes().addAll(provider.getApplicationClassNames());
+    			}	
+            }
+            // Set some soot parameters if not android 
+            else if (!AverroesOptions.isAndroid()){
```
Comment (karimhamdanali) [inline src/averroes/Main.java]: Also here.

File: src/averroes/Main.java [hunk 3]
```diff
@@ -63,29 +68,60 @@ public static void main(String[] args) {
...
+            }
 			Options.v().set_validate(true);
 
 			// Load the necessary classes
 			System.out.println("");
 			System.out.println("Loading classes ...");
 			Scene.v().loadNecessaryClasses();
-			Scene.v().setMainClassFromOptions();
+			
+			//only in case of java files
```
Comment (karimhamdanali) [inline src/averroes/Main.java:124]: either bring the if-body to the same line or surround it with curly braces (even if it's a single statement)

File: src/averroes/options/AverroesOptions.java [hunk 4]
```diff
@@ -306,4 +368,30 @@ public static boolean isApplicationClass(SootClass sootClass) {
...
 	}
-}
\ No newline at end of file
+	
+	/**
+	 * Check if an android app is being processed.
+	 * 
+	 * @return 
+	 */
+		public static boolean isAndroid() {
```
Comment (karimhamdanali) [inline src/averroes/options/AverroesOptions.java]: Clean up the indentation also in the following chunk of code.

File: src/averroes/soot/CodeGenerator.java [hunk 5]
```diff
@@ -515,6 +549,11 @@ private void callFinalize() {
 	 */
 	private void callApplicationMethodsReflectively() {
 		for (SootMethod toCall : getAllMethodsToCallReflectively()) {
+			// TODO: Is that correct? We want to skip lifecycle methods because these are modeled in the dummy main
+			if (isLifeCycle(toCall)) {
```
Comment (karimhamdanali) [inline src/averroes/soot/CodeGenerator.java]: Hmm, why is that support added to callApplicationMethodsReflectively? I'd rather this be added to the proper method and the list of lifecycle methods to be retrieved from the proper list of methods to call.

File: src/averroes/soot/CodeGenerator.java [hunk 6]
```diff
@@ -552,6 +591,11 @@ private void callApplicationMethodsReflectively() {
 		}
 
 	}
+	private boolean isLifeCycle(SootMethod method) {
+		// TODO: Refactor (e.g. store the method signatures inside this class)?
+		AndroidEntryPointUtils m = new AndroidEntryPointUtils();
```
Comment (karimhamdanali) [inline src/averroes/soot/CodeGenerator.java]: What's AndroidEntryPointUtils?

**Review by karimhamdanali — CHANGES_REQUESTED** [review]
Please checkout the comments I added throughout and address them before those changes can be approved.

File: src/averroes/soot/CodeGenerator.java (hunk 6 above)
Comment (eshza) [reply src/averroes/soot/CodeGenerator.java]: Class containing common utility methods for dealing with Android entry points. (part of soot)

Comment (karimhamdanali) [reply src/averroes/soot/CodeGenerator.java]: I see.

**Review by karimhamdanali — APPROVED** [review]

**Pull request merged**